### ISSUE TITLE
Can't use client library for macos arm 64

### ISSUE THREAD

**Issue by alkurop** [davidepianca98/KMQTT#18]
I cloned the repo and published `common` and `client` to maven local.

My project script is just one file
```plugins {
    kotlin("multiplatform") version "1.9.0"
}

group = "me.alex"
version = "1.0-SNAPSHOT"

repositories {
    mavenCentral()
    mavenLocal()
}

kotlin {
    macosArm64("macos") {
        binaries {
            executable {
                entryPoint = "main"
            }
        }
    }

    sourceSets {
        val commonMain by getting {
            dependencies {
                implementation("com.github.davidepianca98:kmqtt-client:0.4.1")
                implementation("com.github.davidepianca98:kmqtt-common:0.4.1")
            }
        }
        val macosMain by getting {
            dependencies {
                implementation("com.github.davidepianca98:kmqtt-common-macosarm64:0.4.1")
                implementation("com.github.davidepianca98:kmqtt-client-macosarm64:0.4.1")
            }
        }
    }
}
```

I have my `main`  function in `src/commonMain/kotlin/Main`

Intellij indexer shows that I can import the MQTT library. Project compilation fails with 
`e: Could not find "openssl-macos-arm64" in [projectDir, /Users/alex/.konan/klib, /Users/alex/.konan/kotlin-native-prebuilt-macos-aarch64-1.9.0/klib/common, /Users/alex/.konan/kotlin-native-prebuilt-macos-aarch64-1.9.0/klib/platform/macos_arm64]
`
1. Is my set up correct?
2. what's the issue with open ssl? What to do here?
3. I also want to build for an other target linuxarm64 (Raspberry Pi). Is this project compatable?

**Comment by davidepianca98** [comment]
Hello, as you can see the file that could not be found is here https://github.com/davidepianca98/KMQTT/tree/master/kmqtt-common/src/nativeInterop and is bundled with the dependencies you included. I suggest trying to invert the lines so that common is before client in the commonMain section.
Also I suggest removing the dependecies of the macosMain section as they get automatically included.
`sourceSets {
    val commonMain by getting {
        dependencies {
            implementation("com.github.davidepianca98:kmqtt-common:0.4.1")
            implementation("com.github.davidepianca98:kmqtt-client:0.4.1")
        }
    }
}`
Let me know if this works.

Yes the project is compatible also with linuxArm64, you just need to include the target in the `kotiln` section.

**Comment by alkurop** [comment]
With suggested set up 
<img width="999" alt="Screenshot 2023-07-25 at 13 59 04" src="https://github.com/davidepianca98/KMQTT/assets/5940685/e2315a03-84cf-43b9-90c0-dfe3cb403675">
<img width="280" alt="Screenshot 2023-07-25 at 13 58 51" src="https://github.com/davidepianca98/KMQTT/assets/5940685/5b9ca3b6-6810-4365-be48-74c097539836">
<img width="928" alt="Screenshot 2023-07-25 at 13 58 37" src="https://github.com/davidepianca98/KMQTT/assets/5940685/39e9068e-2c37-420c-a6a3-1a0dd53a9c77">

**Comment by alkurop** [comment]
I see that `openssl.def` has local paths from your machine.

**Comment by davidepianca98** [comment]
Your final script is correct. Unfortunately the problem is that with Apple targets the openssl klib doesn't get included in the dependency package, as mentioned in #17. I still haven't found a solution for that.

**Comment by alkurop** [comment]
hm, actually I looked once again at the error `e: Could not find "openssl-macos-arm64" in [projectDir, /Users/alex/.konan/klib, /Users/alex/.konan/kotlin-native-prebuilt-macos-aarch64-1.9.0/klib/common, /Users/alex/.konan/kotlin-native-prebuilt-macos-aarch64-1.9.0/klib/platform/macos_arm64] `;
The first place it looks for the is `projectDir` (the actual name is replaced). 

Once I placed the libraries into the root dir It started working, without need to mention `open-ssl` in the script. Safely removed library from `.konan`, cleaned project and it still works. 

Until there's a way to place libs into a folder, keeping them in project dir.
<img width="247" alt="Screenshot 2023-07-25 at 18 29 31" src="https://github.com/davidepianca98/KMQTT/assets/5940685/15e7c96b-a0de-4287-873f-f06342a5e666">
